tools/dependency/release_dates.py

```diff
@@ -38,35 +40,69 @@ def format_utc_date(date):
 def verify_and_print_latest_release(dep, repo, metadata_version, release_date):
     try:
         latest_release = repo.get_latest_release()
-        if latest_release.created_at > release_date and latest_release.tag_name != metadata_version:
-            print(
-                f'*WARNING* {dep} has a newer release than {metadata_version}@<{release_date}>: '
-                f'{latest_release.tag_name}@<{latest_release.created_at}>')
-    except github.UnknownObjectException:
-        pass
+    except github.GithubException as e:
+        print('GithubException {e} while getting latest release.')
+        return
+    if latest_release.created_at > release_date and latest_release.tag_name != metadata_version:
+        print(
+            Fore.YELLOW
+            + f'*WARNING* {dep} has a newer release than {metadata_version}@<{release_date}>: '
+            f'{latest_release.tag_name}@<{latest_release.created_at}>' + Style.RESET_ALL)
 
 
-# Print GitHub release date, throw ReleaseDateError on mismatch with metadata release date.
+# Print GitHub release date, throw ReleaseDateVersionError on mismatch with metadata release date.
 def verify_and_print_release_date(dep, github_release_date, metadata_release_date):
     mismatch = ''
     iso_release_date = format_utc_date(github_release_date)
-    print(f'{dep} has a GitHub release date {iso_release_date}')
+    print(Fore.GREEN + f'{dep} has a GitHub release date {iso_release_date}' + Style.RESET_ALL)
     if iso_release_date != metadata_release_date:
-        raise ReleaseDateError(f'Mismatch with metadata release date of {metadata_release_date}')
+        raise ReleaseDateVersionError(
+            f'Mismatch with metadata release date of {metadata_release_date}')
+
+
+# Extract release date from GitHub API for tagged releases.
+def get_tagged_release_date(repo, metadata_version, github_release):
 
+    try:
+        latest = repo.get_latest_release()
+    except github.GithubException as e:
+        print('GithubException {e} while getting latest release.')
+        latest = ''
+        pass
```

Comment:
this `pass` doesnt do anything

im wondering if it should `return` here - ie if its thrown a `GithubException` im thinking `repo.get_tags` will also fail